domain/secret/state/state.go

```diff
@@ -958,22 +958,45 @@ ON CONFLICT(revision_uuid) DO UPDATE SET
 	return nil
 }
 
-func (st State) upsertSecretValueRef(
+func (st State) insertSecretValueRef(
 	ctx context.Context, tx *sqlair.TX, revisionUUID string, valueRef *coresecrets.ValueRef,
 ) error {
 	insertQuery := `
 INSERT INTO secret_value_ref (*)
+VALUES ($secretValueRef.*)`
+
+	insertStmt, err := st.Prepare(insertQuery, secretValueRef{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = tx.Query(ctx, insertStmt, secretValueRef{
```

Comment:
We can construct a secret value ref instance above and use that to prepare the sql stmt